File: app/playpredict/lib/models/situation.dart
// ignore_for_file: non_constant_identifier_names

import 'package:json_annotation/json_annotation.dart';

part 'situation.g.dart';

@JsonSerializable()
class Situation {
  final int away_score;
  final int home_score;
  final bool is_possessing_team;
  final Position position;
  final int quarter;
  final String time;

  Situation({
    required this.away_score,
    required this.home_score,
    required this.is_possessing_team,
    required this.position,
    required this.quarter,
    required this.time,
  });

  factory Situation.fromJson(Map<String, dynamic> json) =>
      _$SituationFromJson(json);

  Map<String, dynamic> toJson() => _$SituationToJson(this);
}

@JsonSerializable()
class Position {
  final int distance;
  final int down;

  final int yard;

  Position({
    required this.distance,
    required this.down,
    required this.yard,
  });

  factory Position.fromJson(Map<String, dynamic> json) =>
      _$PositionFromJson(json);

  Map<String, dynamic> toJson() => _$PositionToJson(this);
}

File: Server/playpredict/situation.py
import time as t
import json
from . import position


class Situation:

    def __init__(
        self,
        time=t.strptime("15:00", "%M:%S"),
        quarter=1,
        home_score=0,
        away_score=0,
        is_possessing_team=False,
    ):
        self.time = time
        self.quarter = quarter
        self.home_score = home_score
        self.away_score = away_score
        self.position = position.Position()
        self.is_possessing_team = is_possessing_team

    def toJSON(self):
        json_dat = json.dumps(
            self, default=lambda o: o.__dict__, sort_keys=True, indent=4
        )
        # Replace "time" with a formatted string in the form m:SS
        json_obj = json.loads(json_dat)
        json_obj["time"] = t.strftime("%M:%S", self.time)
        new_json_dat = json.dumps(
            json_obj, default=lambda o: o.__dict__, sort_keys=True, indent=4
        )
        return new_json_dat
